codegen/templates/endpoint.tmpl

```diff
@@ -252,16 +252,18 @@ func (w {{$workflow}}) Handle(
 	clientRequest = propagateHeaders{{title .Name}}ClientRequests(clientRequest, reqHeaders)
 	{{end}}
 	clientHeaders := map[string]string{}
-	{{if (ne (len $reqHeaderMapKeys) 0) }}
 	var ok bool
 	var h string
-	{{- end -}}
 	{{range $i, $k := $reqHeaderMapKeys}}
 	h, ok = reqHeaders.Get("{{$k}}")
 	if ok {
 		clientHeaders["{{index $reqHeaderMap $k}}"] = h
 	}
 	{{- end}}
+	h, ok = reqHeaders.Get("X-Test-Override-Service")
+	if ok {
+		clientHeaders["X-Test-Override-Service"] = h
```

Comment:
Since the current client call interface looks like this: `w.Clients.Baz.Compare(ctx, clientHeaders, clientRequest)`, `clientHeaders` seems to be the obvious place to put something to indicate that this is a test request. Maybe I can add another parameter. What do you think?

This header is hardcoded, but it is transparent to any applications. What do you mean by freedom to configure a proper header key?

Reply:
I can change the header to `X-Zanzibar-Use-Staging` so it's more consistent with the per-client configs you suggested, which also hardcodes "staging". Would that be enough?